mlflow/types/utils.py

```diff
@@ -136,6 +141,15 @@ def _infer_schema(data: Any) -> Schema:
                 for field in data.schema.fields
             ]
         )
+    elif isinstance(data, str):
+        schema = Schema([ColSpec(type=DataType.string)])
+    elif isinstance(data, list):
+        if all(isinstance(element, str) for element in data):
+            schema = Schema([ColSpec(type=DataType.string)])
+        elif all(isinstance(element, dict) for element in data) and all(
+            isinstance(value, str) for d in data for value in d.values()
+        ):
```

Comment:
I don't see where we're validating that the dictionary keys are strings. We just seem to be validating values. Can we update this and add test coverage?

Reply:
Added additional validation check, custom exception, and test suite